### PR TITLE
e2e: v2 backup test

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/5367.

### CODE REVIEW

File: e2e/ctl_v2_test.go [hunk 1]
```diff
@@ -227,6 +228,45 @@ func TestCtlV2RoleList(t *testing.T) {
...
 }
 
+func TestCtlV2Backup(t *testing.T) { // For https://github.com/coreos/etcd/issues/5360
+	defer testutil.AfterTest(t)
+
+	backupDir, err := ioutil.TempDir("", "test-backup.etcd")
+	if err != nil {
+		t.Fatal(err)
+	}
+
```
Comment (heyitsanthony) [inline e2e/ctl_v2_test.go]: `defer os.RemoveAll(backupDir)`?

**Comment by gyuho** [comment]
So these are the commands that are run by this test.

> 
./bin/etcd --name testname0 --listen-client-urls http://localhost:20000 --advertise-client-urls http://localhost:20000 --listen-peer-urls http://localhost:20001 --initial-advertise-peer-urls http://localhost:20001 --initial-cluster="testname0=http://localhost:20001" --initial-cluster-token new --data-dir testname0.etcd
./bin/etcdctl --endpoints http://localhost:20000 backup --data-dir testname0.etcd --backup-dir test-backup.etcd
./bin/etcd --name testname0 --listen-client-urls http://localhost:20000 --advertise-client-urls http://localhost:20000 --listen-peer-urls http://localhost:20001 --initial-advertise-peer-urls http://localhost:20001 --initial-cluster="testname0=http://localhost:20001" --initial-cluster-token new --data-dir test-backup.etcd --force-new-cluster

Confirmed that this command panics if I run this manually, without @xiang90's fix.

But somehow when we run on e2e, it's not  passing the message type `ConfChangeRemoveNode`, so doesn't panic even without the Xiang's panic.

Do you see anything that I am doing wrong here?

/cc @xiang90 @heyitsanthony 

Thanks!

File: e2e/ctl_v2_test.go (hunk 1 above)
Comment (gyuho) [inline e2e/ctl_v2_test.go]: Just fixed. Thanks!

File: e2e/etcd_test.go [hunk 2]
```diff
@@ -132,6 +132,7 @@ type etcdProcessConfig struct {
 }
 
 type etcdProcessClusterConfig struct {
+	dataDirPath       string // for single-node cluster
```
Comment (heyitsanthony) [inline e2e/etcd_test.go]: can this be a prefix instead of a path so it can support multiple nodes?

File: e2e/etcd_test.go [hunk 3]
```diff
@@ -402,3 +408,7 @@ func (epc *etcdProcessCluster) endpoints() []string {
 	}
 	return eps
 }
+
+func (epc *etcdProcessCluster) dataDir(i int) string {
```
Comment (heyitsanthony) [inline e2e/etcd_test.go]: I don't think this is necessary since `backends()[0].cfg.dataDirPath` does the same thing

File: e2e/etcd_test.go (hunk 2 above)
Comment (gyuho) [inline e2e/etcd_test.go]: Right. Will fix.

File: e2e/etcd_test.go (hunk 3 above)
Comment (gyuho) [inline e2e/etcd_test.go]: Right. Will fix.

File: e2e/ctl_v2_test.go [hunk 4]
```diff
@@ -227,6 +229,46 @@ func TestCtlV2RoleList(t *testing.T) {
...
+	if err := epc.Close(); err != nil {
+		t.Fatalf("error closing etcd processes (%v)", err)
+	}
+
+	// restart from the backup directory
+	cfg2 := configNoTLS
+	cfg2.dataDirPath = backupDir
+	cfg2.forceNewCluster = true
+	epc2 := setupEtcdctlTest(t, &cfg2, false)
+
```
Comment (heyitsanthony) [inline e2e/ctl_v2_test.go]: this should check if it can get "foo1" so it knows the backup went through

**Comment by gyuho** [comment]
ah, when it starts, it deletes the data dir. That was why it wasn't panic-ing.

Will fix that too.

**Comment by gyuho** [comment]
@heyitsanthony All addressed. PTAL.

Thanks.

File: e2e/etcd_test.go [hunk 5]
```diff
@@ -122,16 +122,22 @@ type etcdProcess struct {
 }
 
 type etcdProcessConfig struct {
-	args        []string
-	dataDirPath string
-	acurl       string
+	args []string
+
+	dataDirPath        string
+	usePreviousDataDir bool
```
Comment (heyitsanthony) [inline e2e/etcd_test.go]: `keepDataDir`?

Comment (gyuho) [inline e2e/etcd_test.go]: fixed.

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
Thanks merging after all greens. Found some other errors.

**Pull request merged**